### ISSUE TITLE
`HexToArrayError` should not wrap `HexToBytesError`

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/hex-conservative#60]
`HexToBytesError` contains odd length variant which is unreachable when parsing array - it's just invalid length.

We should put invalid char directly into `HexToArrayError` and panic during conversion if necessary.

**Comment by tcharding** [comment]
I agree that the issue is correct but posit that the current code is still technically correct.  If fixing as suggested makes the generics any more complicated than they already are then I think it should be left as is.

**Comment by Kixunil** [comment]
I don't see how it has anything to do with generics. Just modify the variants of the enum and panic during conversion. Unless you want to avoid the panic but we can still do that without complicating the *public* API by calling into an inner function after checking the length which only returns errors for invalid characters.

**Comment by tcharding** [comment]
This issue is conceptually correct but not technically correct, currently an odd length string returns the `HexToBytesError::OddLengthString` variant. As can be seen by this existing test:
```rust
        assert_eq!(
            <[u8; 4]>::from_hex(oddlen),
            Err(HexToBytesError::OddLengthString(OddLengthStringError { len: 17 }).into())
        );
```

**Comment by tcharding** [comment]
As I suspected this issue is curly. I've had a bunch of goes at it. I am sick right now though.

**Comment by tankyleo** [comment]
@tcharding I think we can close this issue - it was fixed in #80

**Comment by tcharding** [comment]
Mad, thanks.
